src/foam/nanos/notification/email/SMTPEmailService.js

```diff
@@ -293,6 +316,57 @@ if ( getAuthenticate() ) {
 } else {
   session_ = Session.getInstance(props);
 }`
+    },
+    {
+      name: 'finalizeEmailConfig',
+      javaReturns: 'foam.nanos.notification.email.EmailMessage',
+      args: [
+        {
+          name: 'x',
+          javaType: 'foam.core.X'
+        },
+        {
+          name: 'emailMessage',
+          javaType: 'final foam.nanos.notification.email.EmailMessage'
+        }
+      ],
+      javaCode:
+        `
+foam.nanos.session.Session session = x.get(foam.nanos.session.Session.class);
+
+DAO userDAO         = (DAO) x.get("localUserDAO");
+DAO groupDAO        = (DAO) x.get("groupDAO");
+
+User user           = (User) userDAO.find(session.getUserId());
+Group group         = (Group) groupDAO.find(user.getGroup());
+AppConfig appConfig = group.getAppConfig(x);
+
+String domain = appConfig.getUrl().contains("ablii") ? "ablii.com" : "nanopay.net";
+
+if ( SafetyUtil.isEmpty(emailMessage.getFrom()) ) {
+  emailMessage.setFrom(
+    ! SafetyUtil.isEmpty(group.getFrom()) ?
+      group.getFrom() :
+      getFrom().substring(0, getFrom().indexOf("@") + 1) + domain
```

Comment:
Why would the from not have the correct domain?

Reply:
If _emailMessage.getFrom()_ and _group.getFrom()_ are both empty, we will have to return _SMTPEmailService.getFrom()_.

The getFrom() in the SMTPEmailService will always return the default value which defined in the services journal file(info@nanopay in this case). For example, if the request is from ablii, we have to replace the domain name with ablii.com

Another option is we can set the domain at SessionServerBox (Just like appConfig), then we will always have correct domain for each request.